joomla/joomla-cms
# Search fields SQL query breaks search

### Summary of Changes
Changed SQL query join of #__fields and #__fields_values table to join in proper order.

In original SQL #__fields_values are joined based on item_id, and in WHERE section is condition for fields only with context 'com_content.article'. If category with same id as searched article exists, #__fields_values row for such category will be joined, and the where condition removes the row completely from result. 

Proposed SQL joins first all fields with context 'com_content.article' and rows from #__fields_values only for such fields.

### Testing Instructions

- Create 1 field for com_content.categories context.
- Create 1 category and fill that field value.
- Create 1 article. 
- Make sure that created article has same id as category, if needed create so many articles or categories so there will be 1 article with same id as category.
- Go to frontend and search that article title using com_search.
- apply the patch and search again.

### Expected result
Article should be found.

### Actual result
Article is not found. After applying patch, article is found.
